projects/odf_transform/odf_transform/odf_parser/odf_parser.py

```diff
@@ -199,34 +190,100 @@ def _find_previous_key(key_list, present_key):
                 previous_key = i
         return previous_key
 
-    # Find matching vocabulary
+    # Loop through each variables and detect flag variables
+    for var, att in metadata.items():
+        # Retrieve information from variable name
+        odf_var_name = parse_odf_code_variable(var)
+        # FLAG VARIABLES Detect if it is a flag column and find its related variable
+        flag_column = False
+        related_variable = None
+        if odf_var_name['name'] == 'QQQQ':  # MLI FLAG should apply to previous variable
+            related_variable = _find_previous_key(metadata, var)
```

Comment:
instead of using `_find_previous_key`, would it work if you set a variable `previous_key=var` at the end of the for loop?

Reply:
totally lots has changed now!